### ISSUE TITLE
508-defect-2 [COGNITION, FOCUS MANAGEMENT]: When focus moves to the map, the focus halo should be visible and content announced to screen reader

### ISSUE THREAD

**Issue by jenstrickland** [department-of-veterans-affairs/va.gov-team#17404]
# [508-defect-2](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/accessibility/guidance/defect-severity-rubric.md#508-defect-2)

## Feedback framework

- **❗️ Must** for if the feedback must be applied 
- **⚠️ Should** if the feedback is best practice 
- **✔️ Consider** for suggestions/enhancements 

## Definition of done

1. Review and acknowledge feedback.
1. Fix and/or document decisions made.
1. Accessibility specialist will close ticket after reviewing documented decisions / validating fix.

<hr/>

## Point of Contact

**VFS Point of Contact:** Jennifer

## User Story or Problem Statement

As a user who navigates by keyboard, I expect to see a focus outline as a tab through the screen, so I may orient myself in the content.

## Details

When tabbing through the screen content, moving from the search results to the map, focus moves from the last search result item to the `<canvas>` element for the map, but there is no focus halo. Visible focus indicators help anyone who relies on the keyboard to operate the page, by letting them visually determine the component on which keyboard operations will interact at any point in time. People with attention limitations, short term memory limitations, or limitations in executive processes benefit by being able to discover where the focus is located.

Additionally, using the screen readers, the map is read only as "clickable" (if it is announced at all) when focus is on it also without the focus halo. It would be helpful to have an `aria-label="Search results map area"` on the `<canvas>` element.

## Acceptance Criteria

- [ ] When focus is moved to keyboard operable user interface element, the focus indicator is visible for sighted users.
- [ ] When focus is on the map, the screen reader hears "Search results map area"

## Environment

* Operating System: all
* Browser: all
* Screen reading device: all
* Server destination: staging

## Steps to Recreate

1. Enter `https://staging.va.gov/find-locations/?address=04101&context=Portland%2C%20Maine%2004101%2C%20United%20States&facilityType=health&location=43.66%2C-70.25&page=1&serviceType` in browser
2. Tab through the screen
3. At the last item in the search results, tab to the next, and notice it is not visible where focus is
4. Verify that the map is not announced to the screen reader

## WCAG or Vendor Guidance (optional)

* [WCAG 2.0 Level AA - 2.4.7 Focus Visible](https://www.w3.org/TR/UNDERSTANDING-WCAG20/navigation-mechanisms-focus-visible.html) - Any keyboard operable user interface has a mode of operation where the keyboard focus indicator is visible. 

## Screenshots or Trace Logs

Using `document.activeElement` to uncover where focus is after the last link in the search results, "Mental health: 207-623-8411 x5515" reveals focus is on the `canvas.mapboxgl-canvas' and without a visible indicator of focus. 

![Screen Shot 2020-12-14 at 7 11 35 AM](https://user-images.githubusercontent.com/57469/102080082-d8d67e80-3ddb-11eb-8145-854de349065f.png)

**Comment by brangi** [comment]
merged
